save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt

```diff
@@ -143,10 +149,16 @@ class WarnPlugin(
         // NOTE: SAVE will pass relative paths of Tests (calculated from testRootConfig dir) into the executed tool
         val fileNamesForExecCmd =
                 warnPluginConfig.wildCardInDirectoryMode?.let {
-                    // a hack to put only the directory path to the execution command
+                    // a hack to put only the root directory path to the execution command
                     // only in case a directory mode is enabled
-                    "${copyPaths[0].parent!!}$it"
+                    var testRootPath = testConfig.directory.parent ?: testConfig.directory
+                    while (testRootPath.parent != null && testRootPath.parent != ".".toPath()) {
+                        testRootPath = testRootPath.parent!!
+                    }
+                    "$testRootPath$it"
                 } ?: copyPaths.joinToString(separator = warnPluginConfig.batchSeparator!!)
+
+        logDebug("Constructed file name for execution for warn plugin: $fileNamesForExecCmd")
         val execFlagsAdjusted = resolvePlaceholdersFrom(warnPluginConfig.execFlags, extraFlags, fileNamesForExecCmd)
         val execCmd = "${generalConfig.execCmd} $execFlagsAdjusted"
```

Comment:
Pls create an issue and assign it on me for running execCmd once

Reply:
https://github.com/analysis-dev/save/issues/350